src/implementations/unwrap/unwrap.ts

```diff
@@ -0,0 +1,78 @@
+import { documentsInDirectory, readOpenAttestationFile, writeDocumentToDisk } from "../utils/disk";
+import mkdirp from "mkdirp";
+import { getData, OpenAttestationDocument } from "@govtechsg/open-attestation";
+import path from "path";
+
+export enum Output {
+  File,
+  Directory,
+  StdOut,
+}
+
+export const unwrapIndividualDocuments = async (
+  wrappedDocumentPath: string,
+  unwrappedDocumentDir: string | undefined,
+  outputPathType: Output
+): Promise<OpenAttestationDocument[]> => {
+  const oaArray: OpenAttestationDocument[] = [];
+  const documentFileNames = await documentsInDirectory(wrappedDocumentPath);
+
+  for (const file of documentFileNames) {
+    const unwrappedDocument = getData(readOpenAttestationFile(file));
+    oaArray.push(unwrappedDocument);
+
+    try {
+      // Write unwrapped document to new directory
+      writeOutput({
+        outputPathType,
+        unwrappedDocumentPath: unwrappedDocumentDir,
+        file,
+        document: unwrappedDocument,
+      });
+    } catch (e) {
+      throw e;
+    }
+  }
+  return oaArray;
+};
+
+const writeOutput = ({
+  outputPathType,
+  unwrappedDocumentPath,
+  file,
+  document,
+}: {
+  outputPathType: Output;
+  unwrappedDocumentPath?: string;
+  file: string;
+  document: any;
+}): void => {
+  if (outputPathType === Output.File && unwrappedDocumentPath) {
+    writeDocumentToDisk(path.parse(unwrappedDocumentPath).dir, path.parse(unwrappedDocumentPath).base, document);
+  } else if (outputPathType === Output.Directory && unwrappedDocumentPath) {
+    writeDocumentToDisk(unwrappedDocumentPath, path.parse(file).base, document);
+  } else {
+    console.log(JSON.stringify(document, undefined, 2)); // print to console, no file created
+  }
+};
```

Comment:
refactor this and the one in implementation/wrap.ts, (move into disk utils) then consume here.

Reply:
Shifted to [`src/implementations/utils/disk.ts`](https://github.com/Open-Attestation/open-attestation-cli/blob/feat/add-unwrap-command/src/implementations/utils/disk.ts#L43)

Reply:
i meant the whole `writeOutput` function

Reply:
Shifted `writeOutput` function to [`disk.ts` ](https://github.com/Open-Attestation/open-attestation-cli/blob/feat/add-unwrap-command/src/implementations/utils/disk.ts#L48)